src/mozilla_sec_eia/models/sec10k/sec_output_table.py

```diff
@@ -0,0 +1,367 @@
+"""Module for creating an SEC 10K output table with filing companies and subsidiary companies."""
+
+import logging
+import re
+from importlib import resources
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from dagster import AssetIn, AssetOut, multi_asset
+
+from mozilla_sec_eia.models.sec10k.utils.cloud import (
+    convert_ex21_id_to_filename,
+)
+from mozilla_sec_eia.models.sec_eia_record_linkage.preprocessing import (
+    company_name_cleaner,
+)
+
+from .extract import (
+    sec10k_filing_metadata,
+    year_quarter_partitions,
+)
+
+logger = logging.getLogger(f"catalystcoop.{__name__}")
+
+INVALID_NAMES = [
+    "llc",
+    "limited liability company",
+    "limited",
+    "ltd",
+    "iiii",
+    "inc",
+    "incorporated",
+    "partnership",
+    "i",
+    "name",
+    "company",
+    "&",
+    "",
+]
+
+
+def _remove_weird_sec_cols(sec_df) -> pd.DataFrame:
+    weird_cols = ["]fiscal_year_end", "]irs_number", "]state_of_incorporation"]
+    for weird_col in weird_cols:
+        if weird_col not in sec_df:
+            continue
+        normal_col = weird_col[1:]
+        sec_df.loc[:, normal_col] = sec_df[normal_col].where(
+            sec_df[weird_col].isnull(), sec_df[weird_col]
+        )
+        sec_df = sec_df.drop(columns=[weird_col])
+    return sec_df
+
+
+def _add_report_year_to_sec(sec_df: pd.DataFrame, md: pd.DataFrame) -> pd.DataFrame:
+    """Merge metadata on to get a report year for extracted SEC data.
+
+    Expects filename to be the index of the SEC dataframe.
+    """
+    sec_df = sec_df.merge(md[["filename", "date_filed"]], how="left", on=["filename"])
+    sec_df = sec_df.rename(columns={"date_filed": "report_date"})
+    sec_df.loc[:, "report_year"] = (
+        sec_df["report_date"].astype("datetime64[ns]").dt.year
+    )
+    return sec_df
+
+
+def _flatten_sec_companies_across_time(sec_df: pd.DataFrame) -> pd.DataFrame:
+    """Keep only the most recent record for each unique SEC company.
+
+    Note that this drops old records for companies that have changed
+    names or addresses across time. Also, we group by sec_company_id not
+    CIK, so filer companies and subsidiary companies are unique in the
+    output dataframe.
+    TODO: create an asset that tracks name and address chnages across
```

Comment:
I left a comment about how we should probably have another asset that has records for each company and report year, and tracks changes across time. I guess this would be easy to make from this module and is maybe worth just putting in now